numpy/lib/tests/test_io.py

```diff
@@ -1827,11 +1827,11 @@ def test_empty_file(self):
             sup.filter(message="genfromtxt: Empty input file:")
             data = TextIO()
             test = np.genfromtxt(data)
-            assert_equal(test, np.array([]))
```

Comment:
Here, we would have a mismatch between `np.array([], dtype=[])` and `np.array([], dtype=float64)`